mitodl/hugo-course-publisher
# search proof of concept

#### Pre-Flight checklist


#### What are the relevant tickets?

closes #158

#### What's this PR do?

this adds a search UI, built with React, based on our existing UI in open-discussions. 

#### How should this be manually tested?

You'll need to do a little  bit of setup.

First, you have to have open-discussions running locally. You'll need to add the origin for ocw-next to discussions' CORS allowlist so that the OCW frontend can make requests to open's backend. Adding this to open's `.env` worked for me:

```
OPEN_DISCUSSIONS_CORS_ORIGIN_WHITELIST=['localhost:3000']
```

then you should be able to install npm packages and check out `/search`. You'll also need to add a `.env` file in the `ocw-to-hugo` directory which looks like this:

```js
SEARCH_API_URL=http://localhost:8063/api/v0/search/
```

Then you should be able to try it out. You'll need to have your local open-discussions running, and you'll need to have run the `backpopulate_ocw_data` management command. 

There should be a search box on the homepage. If you type in a query it will then take you to `/search` with that query set in the URL bar. If you're already on search you should be able to change the search and it will be-run.

#### Screenshots (if appropriate)

![Screenshot from 2020-08-10 16-15-29](https://user-images.githubusercontent.com/6207644/89826782-c514c600-db24-11ea-93d8-4e658e67161a.png)

![Screenshot from 2020-08-10 16-15-41](https://user-images.githubusercontent.com/6207644/89826791-c6de8980-db24-11ea-9678-0be99071d0ac.png)
